orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/manifest/ManifestEvaluator.java

```diff
@@ -16,172 +16,245 @@
   private static final ThreadLocal<Yaml> yamlParser =
       ThreadLocal.withInitial(() -> new Yaml(new SafeConstructor()));
 
   private final ArtifactUtils artifactUtils;
-  private final OortService oort;
-  private final ObjectMapper objectMapper;
   private final ContextParameterProcessor contextParameterProcessor;
-  private final KatoService kato;
+  private final KatoService katoService;
+  private final ObjectMapper objectMapper;
+  private final OortService oortService;
+  private final RetrySupport retrySupport;
 
-  private final RetrySupport retrySupport = new RetrySupport();
+  @Autowired
+  public ManifestEvaluator(
+      ArtifactUtils artifactUtils,
+      ContextParameterProcessor contextParameterProcessor,
+      KatoService katoService,
+      ObjectMapper objectMapper,
+      OortService oortService,
+      RetrySupport retrySupport) {
+    this.artifactUtils = artifactUtils;
+    this.contextParameterProcessor = contextParameterProcessor;
+    this.katoService = katoService;
+    this.objectMapper = objectMapper;
+    this.oortService = oortService;
+    this.retrySupport = retrySupport;
+  }
 
   @RequiredArgsConstructor
   @Getter
   public static class Result {
-    private final List<Map<Object, Object>> manifests;
-    private final List<Artifact> requiredArtifacts;
-    private final List<Artifact> optionalArtifacts;
+    private final ImmutableList<Map<Object, Object>> manifests;
+    private final ImmutableList<Artifact> requiredArtifacts;
+    private final ImmutableList<Artifact> optionalArtifacts;
   }
 
+  /**
+   * Resolves manifests and associated artifacts as a {@link Result}. Handles determining the input
+   * source of a manifest, downloading the manifest if it is an artifact, optionally processing it
+   * for SpEL, and composing the required and optional artifacts associated with the manifest.
+   *
+   * @param stage The stage to consider when resolving manifests and artifacts.
+   * @param context The stage-specific manifest context to consider when resolving manifests and
+   *     artifacts.
+   * @return The result of the manifest and artifact resolution {@link Result}.
+   */
   public Result evaluate(Stage stage, ManifestContext context) {
-    Iterable<Object> rawManifests;
-    List<Map<Object, Object>> manifests = Collections.emptyList();
+    return new Result(
+        getManifests(stage, context),
+        getRequiredArtifacts(stage, context),
+        getOptionalArtifacts(stage));
+  }
+
+  private ImmutableList<Map<Object, Object>> getManifests(Stage stage, ManifestContext context) {
     if (ManifestContext.Source.Artifact.equals(context.getSource())) {
-      Artifact manifestArtifact =
-          artifactUtils.getBoundArtifactForStage(
-              stage, context.getManifestArtifactId(), context.getManifestArtifact());
+      return getManifestsFromArtifact(stage, context);
+    }
+    return getManifestsFromText(context);
```

Comment:
Personally I'd lean towards a switch for `enum` types since it makes it a little more clear to readers what types you're handling and makes it less likely to have strange things happen if you ever add a third type:
```
    switch (context.getSource()) {
      case Artifact:
        return getManifestsFromArtifact(stage, context);
      case Text:
        return getManifestsFromText(context);
    }
    throw new IllegalStateException("Unknown ManifestContext.Source " + context.getSource());
```

Another bonus of this is static analysis could catch at compile time that you're not covering all the cases... internally this code would no longer compile if someone added a third enum value, but we don't have that set up in Spinnaker. Maybe some day, though!

Reply:
:1234: :tada: :+1: